lib/install/management/dispatcher.go

```diff
@@ -245,3 +254,47 @@ func (d *Dispatcher) CollectDiagnosticLogs() {
 		}
 	}
 }
+
+func (d *Dispatcher) DeleteVCH(conf *metadata.VirtualContainerHostConfigSpec) error {
+	log.Infof("Removing VMs")
+
+	if err := d.DeleteVCHInstances(conf); err != nil {
+		return err
+	}
+	if err := d.destroyResourcePoolIfEmpty(conf); err != nil {
+		return err
+	}
+	return d.removeNetwork(conf)
+}
+
+func (d *Dispatcher) DeleteVCHInstances(conf *metadata.VirtualContainerHostConfigSpec) error {
+	var errs []string
+
+	var err error
+	var vmm *vm.VirtualMachine
+	var children []*vm.VirtualMachine
+
+	if vmm, err = d.findAppliance(conf); err != nil {
```

Comment:
This should be `d.VMFromID(id string)` or similar and should use the encoded moref, not the name:
```
moref := new(types.ManagedObjectReference)
ok := moref.FromString(id)
if !ok {
 ...
}
ref, err := v.Session.Finder.ObjectReference(ctx, *moref)
vm, ok := ref.(*object.VirtualMachine)
if !ok {
  ...
}
```